src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultDocumentSnapshot.cs

```diff
@@ -88,36 +88,42 @@ public override bool TryGetGeneratedOutput(out RazorCodeDocument result)
         {
             if (State.IsGeneratedOutputResultAvailable)
             {
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
```

Comment:
nit: Might help readers if you include a comment explaining why this makes sense. The other option is to use `[SuppressMessage("...", "VSTHRD002:...", Justification = "...")]` above the method, including a real `Justification`.